### ISSUE TITLE
Protocol and mixin inheritance order

### ISSUE THREAD

**Issue by adusamr** [microsoft/pyright#10666]
```python
class MyProtocol(Protocol):
    foo: int

class FooPropery:
    @property
    def foo(self: MyProtocol) -> int:
        return 0

@final
class MyClass1(MyProtocol, FooPropery): # "MyProtocol.foo" is not implemented
    def __init__(self) -> None:
        pass

@final
class MyClass2(FooPropery, MyProtocol): # no error
    def __init__(self) -> None:
        pass
```
Inheritance order affects checking of implementation of a protocol.
In the above code, only `MyClass1` shows an error.

This is not specific to property. Replacing `FooProperty` to below `FooSetter` I still get the same result.
```python
class FooSetter:
    def set_foo(self: MyProtocol, foo: int):
        self.foo = foo
```

**VS Code extension or command-line**
VS Code extension Pylance version: 2025.6.2

**Comment by erictraut** [comment]
Both of these examples incorrectly override `foo`. I'm guessing that you've suppressed the `reportIncompatibleVariableOverride` check. If you enable this check, you'll see that an error is generated for both classes.

The `MyClass1` also attempts to override `MyProtocol` class and fails to implement the abstract attribute `foo`, which explains why you see a `reportGeneralTypeIssues` diagnostic reported in this case. This behavior is covered by the typing spec, and pyright is conformant with the spec. For more details, refer to [this section](https://typing.python.org/en/latest/spec/protocol.html#merging-and-extending-protocols). Mypy generates the same error in this case — and likewise does not generate this error for MyClass2. 

So, I think pyright is working correctly here.
